From 97b0ebce48ad1ad218919ec76c853f8696b22658 Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Wed, 6 Mar 2024 17:49:06 +0100
Subject: [PATCH 0911/1141] ARM: dts: stm32: simplify the OPP description for
 STM32MP15
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Move the OPP description at Soc Level, as the supported OPP are managed
dynamically by opp-supported-hw value.

The empty files "stm32mp151xa.dtsi" are kept only for compatibility
with device tree generated by tool STM32CubeMX.

Even if this patch allows to boot a D/F device tree variant on board
with a A/C SoC with the expected OPP, the critical threshold temperature
for thermal protection for 800MHz variant is used (105°C instead of 120°C).

So it is still recommended to include the correct SoC
dtsi files (stm32mp15xa.dtsi, stm32mp15xc.dtsi, stm32mp15xd.dtsi or
stm32mp15xf.dtsi) in your board device tree to have the expected
thermal protection for your "Mission Profile" with the correct
CortexA35 frequency.

Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: I47a1825371d306e5aa432db3d328e7c722ddacdd
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/365050
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 arch/arm/boot/dts/stm32mp151.dtsi  | 19 +++++++++++++++++++
 arch/arm/boot/dts/stm32mp15xa.dtsi |  8 --------
 arch/arm/boot/dts/stm32mp15xd.dtsi | 14 --------------
 3 files changed, 19 insertions(+), 22 deletions(-)

--- a/arch/arm/boot/dts/stm32mp151.dtsi
+++ b/arch/arm/boot/dts/stm32mp151.dtsi
@@ -32,6 +32,25 @@
 	cpu0_opp_table: cpu0-opp-table {
 		compatible = "operating-points-v2";
 		opp-shared;
+
+		opp-400000000 {
+			opp-hz = /bits/ 64 <400000000>;
+			opp-microvolt = <1200000>;
+			opp-supported-hw = <0x2>;
+			opp-suspend;
+		};
+
+		opp-650000000 {
+			opp-hz = /bits/ 64 <650000000>;
+			opp-microvolt = <1200000>;
+			opp-supported-hw = <0x1>;
+		};
+
+		opp-800000000 {
+			opp-hz = /bits/ 64 <800000000>;
+			opp-microvolt = <1350000>;
+			opp-supported-hw = <0x2>;
+		};
 	};
 
 	intc: interrupt-controller@a0021000 {
--- a/arch/arm/boot/dts/stm32mp15xa.dtsi
+++ b/arch/arm/boot/dts/stm32mp15xa.dtsi
@@ -3,11 +3,3 @@
  * Copyright (C) STMicroelectronics 2022 - All Rights Reserved
  * Author: Alexandre Torgue <alexandre.torgue@foss.st.com> for STMicroelectronics.
  */
-
-&cpu0_opp_table {
-		opp-650000000 {
-			opp-hz = /bits/ 64 <650000000>;
-			opp-microvolt = <1200000>;
-			opp-supported-hw = <0x1>;
-		};
-};
--- a/arch/arm/boot/dts/stm32mp15xd.dtsi
+++ b/arch/arm/boot/dts/stm32mp15xd.dtsi
@@ -4,20 +4,6 @@
  * Author: Alexandre Torgue <alexandre.torgue@foss.st.com> for STMicroelectronics.
  */
 
-&cpu0_opp_table {
-		opp-800000000 {
-			opp-hz = /bits/ 64 <800000000>;
-			opp-microvolt = <1350000>;
-			opp-supported-hw = <0x2>;
-		};
-		opp-400000000 {
-			opp-hz = /bits/ 64 <400000000>;
-			opp-microvolt = <1200000>;
-			opp-supported-hw = <0x2>;
-			opp-suspend;
-		};
-};
-
 &cpu_thermal {
 	trips {
 		cpu-crit {
